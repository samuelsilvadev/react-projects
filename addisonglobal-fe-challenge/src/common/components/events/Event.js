import React from 'react';
import PropTypes from 'prop-types';

import './Event.scss';

import Market from './Market';

const renderMarket = (market) => (
	<li className="event__market__item" key={market.id}>
		<Market {...market} />
	</li>
);

const Event = ({ name, markets = [] }) => {
	return (
		markets.length > 0 && (
			<section className="event">
				<h3 className="event__title">{name}</h3>
				<ul className="event__market-wrapper">
					{markets.map(renderMarket)}
				</ul>
			</section>
		)
	);
};

Event.propTypes = {
	name: PropTypes.string,
	markets: PropTypes.arrayOf(PropTypes.object)
};

export default Event;
