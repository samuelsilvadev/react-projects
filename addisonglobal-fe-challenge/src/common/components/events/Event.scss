@import '../../scss/main.scss';

.event {
	border: 2px solid $fifth;

	&:not(:last-of-type) {
		margin-bottom: 20px;
	}

	&__title {
		font-family: 'Lato', sans-serif;
		font-size: 17px;
		font-weight: 700;
		text-align: center;
		padding: 23px 0;
	}

	&__market-wrapper {
		list-style: none;
	}

	&__market__item {
		border-top: 2px solid $fifth;
		padding: 10px;
	}
}
