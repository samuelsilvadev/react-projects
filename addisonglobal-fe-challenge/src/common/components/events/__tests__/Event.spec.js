import React from 'react';
import { render } from '@testing-library/react';

import Event from '../Event';

jest.mock('../../../contexts/AsideContext', () => ({
	useAsideContext() {
		return {
			handleOpen: () => {}
		};
	}
}));

jest.mock('../../market/Market', () => () => null);

describe('<Event />', () => {
	it('should render correctly with a event', () => {
		const props = {
			name: 'Brazil x Germany',
			markets: [
				{
					id: 1,
					name: 'One'
				}
			]
		};
		const { container, queryByText } = render(<Event {...props} />);

		expect(queryByText(props.name)).toBeVisible();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not render when has no `markets`', () => {
		const props = {
			name: 'Spain x Belgium',
			markets: []
		};
		const { container, queryByText } = render(<Event {...props} />);

		expect(queryByText(props.name)).not.toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not break if market be `undefined`', () => {
		const props = {
			name: 'Spain x Belgium 2',
			markets: undefined
		};
		const { container, queryByText } = render(<Event {...props} />);

		expect(queryByText(props.name)).not.toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});
});
