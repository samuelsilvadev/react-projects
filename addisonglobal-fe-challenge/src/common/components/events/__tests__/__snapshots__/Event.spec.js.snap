// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should not break if market be \`undefined\` 1`] = `null`;

exports[`<Event /> should not render when has no \`markets\` 1`] = `null`;

exports[`<Event /> should render correctly with a event 1`] = `
<section
  class="event"
>
  <h3
    class="event__title"
  >
    Brazil x Germany
  </h3>
  <ul
    class="event__market-wrapper"
  >
    <li
      class="event__market__item"
    />
  </ul>
</section>
`;
